store/tikv/mock-tikv/mvcc.go

```diff
@@ -166,17 +166,22 @@ func (e *mvccEntry) Commit(startTS, commitTS uint64) error {
 
 func (e *mvccEntry) Rollback(startTS uint64) error {
 	if e.lock == nil || e.lock.startTS != startTS {
-		if commitTS, ok := e.checkTxnCommitted(startTS); ok {
-			return ErrAlreadyCommitted(commitTS)
+		c, ok := e.getTxnCommitInfo(startTS)
+		if ok && c.valueType != typeRollback {
+			return ErrAlreadyCommitted(c.commitTS)
 		}
-		return nil
+		if ok && c.valueType == typeRollback {
+			return nil
+		}
+	} else {
+		e.lock = nil
 	}
+
 	e.values = append([]mvccValue{{
 		valueType: typeRollback,
 		startTS:   startTS,
 		commitTS:  startTS,
 	}}, e.values...)
-	e.lock = nil
```

Comment:
No, we shouldn't clean the lock if the lock belongs to another transaction. @hhkbp2 .
This may happens in the following case:
1. txn1 (t1) finished prewrite A, now A has lock(A,t1)
2. txn2(t2) prewrite(A,B), B failed
3. txn2(t2)  try to rollback A. here we write  a commit(A,t2,rollback).
4. txn1(t1) commit A could be success. Here `txn1` shouldn't be influenced by` txn2`.